ml-agents/mlagents/trainers/torch/networks.py

```diff
@@ -280,15 +286,21 @@ def forward(
             vec_inputs, vis_inputs, memories=memories, sequence_length=1
         )
 
-        # TODO: How this is written depends on how the inference model is structured
-        action_out = self.action_model.get_action_out(encoding, masks)
-        return (
-            action_out,
+        cont_action_out, disc_action_out, action_out_deprecated = self.action_model.get_action_out(
+            encoding, masks
+        )
+        export_out = [
+            action_out_deprecated,
             self.version_number,
             torch.Tensor([self.network_body.memory_size]),
-            self.is_continuous_int,
-            self.act_size_vector,
-        )
+            self.is_continuous_int_deprecated,
+            self.act_size_vector_deprecated,
+        ]
+        if self.action_spec.continuous_size > 0:
```

Comment:
The size of the return tuple here varies, so the type hint is incorrect (or really, you shouldn't be using a tuple)

Reply:
I understand it doesn't look good but this is directly used by `torch.onnx.export()` and has to fit in its expected format.

At this moment torch.onnx.export() doesn't accept `None` as tensor to be exported, so this return has to be variable length. I saw related request to pytorch but they haven't added support to this yet.

I tested with List of tensors and it seems to work too. I didn't want to change the type since this is the what the export function usually takes but I can do that if that's better.

I'm not sure if there's better solution than disabling typing here.

Reply:
I don't think tuple vs list matters much. 

I think you can make the return type `List[Union[int, torch.Tensor]]` as long as you don't get complaints about conflicts with existing signatures in the base classes. If all else fails, I guess `# type: ingore` will bypass all the checks here.

Please add some of this as a comment in the code.